stdlib/importlib/abc.pyi

```diff
@@ -82,24 +91,82 @@ if sys.version_info >= (3, 7):
         def contents(self) -> Iterator[str]: ...
 
 if sys.version_info >= (3, 9):
+    _P = TypeVar("_P", bound=Traversable)
     @runtime_checkable
     class Traversable(Protocol):
         @abstractmethod
-        def iterdir(self) -> Iterator[Traversable]: ...
+        def is_dir(self) -> bool: ...
         @abstractmethod
-        def read_bytes(self) -> bytes: ...
+        def is_file(self) -> bool: ...
         @abstractmethod
-        def read_text(self, encoding: Optional[str] = ...) -> str: ...
+        def iterdir(self: _P) -> Generator[_P, None, None]: ...
         @abstractmethod
-        def is_dir(self) -> bool: ...
+        def joinpath(self: _P, child: StrPath) -> _P: ...
+        # Adapted from builtins.open
+        # Text mode: always returns a TextIOWrapper
```

Comment:
Instead of copy/pasting the comments, maybe just mention that this is same as in `pathlib.pyi`, and add a comment to `pathlib.pyi` that `importlib/abc.pyi` must be kept in sync?

Reply:
Added!